morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -250,6 +259,38 @@ public void ensureCaps() {
     }
 
     @Override
+    public void enableDocumentValidation() {
+        for (final MappedClass mc : mapper.getMappedClasses()) {
```

Comment:
Just to confirm: is the list of mapped classes essentially immutable at this point?